packages/bento-design-system/src/Table/Table.tsx

```diff
@@ -398,29 +398,35 @@ function ColumnHeader<D extends Record<string, unknown>>({
       style={{ ...style, zIndex: sticky ? zIndexes.leftStickyHeader : zIndexes.header }}
     >
       <Box
-        className={columnHeaderRecipe({ firstColumn: first, lastColumn: last })}
+        className={columnHeader}
         background={config.headerBackgroundColor}
         color={config.headerForegroundColor}
         {...column.getHeaderProps(column.getSortByToggleProps())}
         textAlign={column.align}
+        {...config.padding.header}
       >
-        {hasHeaderContent && (
-          <Columns space={8} alignY="center" align={column.align}>
-            {column.Header ? (
-              <Column width="content">
-                <Label size="large">{column.render("Header") as any}</Label>
-              </Column>
-            ) : null}
-            {hint && <Column width="content">{hint}</Column>}
-            {sortIcon && (
-              <Column width="content">
-                <Box className={sortIconContainer}>
-                  {sortIcon({ size: 8, color: "currentColor" })}
-                </Box>
-              </Column>
-            )}
-          </Columns>
-        )}
+        <Box
+          paddingLeft={first ? config.boundaryPadding : 0}
+          paddingRight={last ? config.boundaryPadding : 0}
```

Comment:
minor, but I thing it makes more sense to use `undefined` instead of setting the padding to `0` explicitly

```suggestion
          paddingLeft={first ? config.boundaryPadding : undefined}
          paddingRight={last ? config.boundaryPadding : undefined}
```